### PR TITLE
(docs) Add supported units to EXTRACT.

### PR DESCRIPTION
**Pull request by msudberg**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6287)

### CODE REVIEW

File: doc/user/content/sql/functions/extract.md [hunk 1]
```diff
@@ -16,17 +16,56 @@ Parameter | Type | Description
 ----------|------|------------
 _val_ | [`date`](../../types/date), [`timestamp`](../../types/timestamp), [`timestamp with time zone`](../../types/timestamptz) | The value from which you want to extract a component.
 
+### Arguments
+
+`EXTRACT` supports multiple synonyms for most of time periods.
```
Comment (benesch) [inline doc/user/content/sql/functions/extract.md:21]: That "of" seems stray to me?

**Review by benesch — APPROVED** [review]
Great!

**Review by awang — APPROVED** [review]
the change itself lgtm

some observations:
1) the SQL functions page (sql/functions/) _really_ needs a nav or TOC
2) The EXTRACT page doesn't seem to be linked anywhere 
3) Interestingly, the jsonb* type pages 404 from the Netlify preview.  It works fine on the actual docs pages, so I don't think there's an actual issue:  https://deploy-preview-6287--materializeinc.netlify.app/sql/functions/#json-func

File: doc/user/content/sql/functions/extract.md [hunk 2]
```diff
@@ -16,17 +16,56 @@ Parameter | Type | Description
 ----------|------|------------
 _val_ | [`date`](../../types/date), [`timestamp`](../../types/timestamp), [`timestamp with time zone`](../../types/timestamptz) | The value from which you want to extract a component.
 
+### Arguments
+
+`EXTRACT` supports multiple synonyms for most time periods.
```
Comment (awang) [inline doc/user/content/sql/functions/extract.md:21]: suggestion, feel free to ignore:  wdyt of 

> `EXTRACT` supports multiple synonyms for most time periods that can be used interchangeably.

I know it's redundant, but the statement feels a little terse, and for some reason my brain had to think a little bit when i first read it

**Review by awang — APPROVED** [review]

**Comment by msudberg** [comment]
> 1. the SQL functions page (sql/functions/) _really_ needs a nav or TOC

This is what @sploiselle and I were talking about last week when you joined us. It's complicated to implement because the functions part of the page is compiled from other files in a way that makes the default right-nav feature not work for it.

File: doc/user/content/sql/functions/extract.md (hunk 1 above)
Comment (msudberg) [reply doc/user/content/sql/functions/extract.md]: Fixed.

**Comment by msudberg** [comment]
> 2. The EXTRACT page doesn't seem to be linked anywhere

It's linked from the summary for `EXTRACT` on http://localhost:1313/sql/functions/. That's currently the only way we're linking to any of the functions doc. I'd like to change that, but it's too big an architectural issue to be combined with this ticket.

File: doc/user/content/sql/functions/extract.md (hunk 2 above)
Comment (msudberg) [reply doc/user/content/sql/functions/extract.md:21]: Does 

> `EXTRACT` supports the use of different synonyms to denote most time periods.

work better?

**Comment by sploiselle** [comment]
> > 1. the SQL functions page (sql/functions/) _really_ needs a nav or TOC
> 
> This is what @sploiselle and I were talking about last week when you joined us. It's complicated to implement because the functions part of the page is compiled from other files in a way that makes the default right-nav feature not work for it.

Yeah, exactly right.

I also don't think this is as large of a usability hindrance as one might imagine; a lot of folks will just CMD+F for the function name they want. This is how I primarily finds functions in the Postgres docs. The bigger concern is that the functions we support are not findable in search whatsoever, TOC or no.

File: doc/user/content/sql/functions/extract.md (hunk 2 above)
Comment (awang) [reply doc/user/content/sql/functions/extract.md:21]: 👍 thank you!

**Pull request merged**